static/app/views/issueDetails/groupTags.tsx

```diff
@@ -1,195 +1,167 @@
-    return {
-      ...super.getDefaultState(),
-      tagList: null,
-      now: Date.now(),
-    };
-  }
-
-  getEndpoints(): ReturnType<DeprecatedAsyncComponent['getEndpoints']> {
-    const {group, environments, organization, event} = this.props;
-
-    if (
-      organization.features.includes('performance-duration-regression-visible') &&
-      group.issueType === IssueType.PERFORMANCE_DURATION_REGRESSION &&
-      this.state
-    ) {
-      if (!event) {
-        // We need the event for its occurrence data to get the timestamps
-        // for querying
-        return [];
-      }
-
-      const {transaction, aggregateRange2, breakpoint} =
-        event.occurrence?.evidenceData ?? {};
-      return [
-        [
-          'tagFacets',
-          `/organizations/${organization.slug}/events-facets/`,
-          {
-            query: {
-              environment: environments,
-              transaction,
-              includeAll: true,
-              query: getSampleEventQuery({
-                transaction,
-                durationBaseline: aggregateRange2,
-                addUpperBound: false,
-              }),
-              start: new Date(breakpoint * 1000).toISOString(),
-              end: new Date(this.state.now).toISOString(),
-            },
-          },
-        ],
-      ];
-    }
-
-    return [
-      [
-        'tagList',
-        `/organizations/${organization.slug}/issues/${group.id}/tags/`,
-        {
-          query: {environment: environments},
-        },
-      ],
-    ];
-  }
+type SimpleTag = {
+  key: string;
+  topValues: Array<{
+    count: number;
+    name: string;
+    value: string;
+    query?: string;
+  }>;
+  totalValues: number;
+};
 
-  componentDidUpdate(prevProps: Props) {
-    if (
-      !isEqual(prevProps.environments, this.props.environments) ||
-      !isEqual(prevProps.event, this.props.event)
-    ) {
-      this.remountComponent();
-    }
-  }
+function isTagFacetsResponse(
+  _: GroupTagsResponse | Tag[] | undefined,
+  shouldUseTagFacetsEndpoint: boolean
+): _ is Tag[] {
+  return shouldUseTagFacetsEndpoint;
+}
 
-  onRequestSuccess({stateKey, data}: {data: any; stateKey: string}): void {
-    if (stateKey === 'tagFacets') {
-      this.setState({
-        tagList: data
-          .filter(({key}) => key !== 'transaction')
-          .map(sumTagFacetsForTopValues),
-        tagFacets: undefined,
-      });
-    }
+function GroupTags({group, baseUrl, environments, event}: GroupTagsProps) {
+  const organization = useOrganization();
+  const location = useLocation();
+  const now = useMemo(() => Date.now(), []);
+
+  const {transaction, aggregateRange2, breakpoint} =
+    event?.occurrence?.evidenceData ?? {};
+
+  const shouldUseTagFacetsEndpoint =
+    organization.features.includes('performance-duration-regression-visible') &&
+    defined(event) &&
+    group.issueType === IssueType.PERFORMANCE_DURATION_REGRESSION;
+
+  const {data, isLoading, isError, refetch} = useFetchIssueTags({
```

Comment:
maybe just add a default here?
```suggestion
  const {data = [], isLoading, isError, refetch} = useFetchIssueTags({
```